persistent/persistence.py

```diff
@@ -95,12 +98,18 @@ def _get_oid(self):
     def _set_oid(self, value):
         if value == self.__oid:
             return
-        if value is not None:
-            if not isinstance(value, OID_TYPE):
-                raise ValueError('Invalid OID type: %s' % value)
-        if self.__jar is not None and self.__oid is not None:
-            raise ValueError('Already assigned an OID by our jar')
-        self.__oid = value
+        # The C implementation allows *any* value to be
+        # used as the _p_oid.
```

Comment:
If we want to have the C version implement these checks and restore them to the Python version ( and presumably add the length check both places), I'm happy to do that work. But I suggest it be a different PR, after this one is merged and released, because it is a breaking change that requires a new major version number IMO. 

This PR is intended to be a minor release just bringing better C/Python compatibility. I do feel that removing the check from the Python version in this PR is the right thing to do because it makes the C and Python implementations behave the same way, and existing code does depend on the C behaviour.